datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -1677,14 +1677,20 @@ pub fn calculate_union(
 ) -> Result<EquivalenceProperties> {
     // TODO: In some cases, we should be able to preserve some equivalence
     //       classes. Add support for such cases.
-    let mut init = eqps[0].clone();
+    let mut iter = eqps.into_iter();
```

Comment:
the new version also will return an error rather than panic if no equivalence properties are returned